### ISSUE TITLE
buffer_get_var does not get b:changedtick value

### ISSUE THREAD

**Issue by ZyX-I** [neovim/neovim#5131]
At 67062d20ede952f0658f2418867adce381a69c64 if you run `curbufmeths.get_var('changedtick')` in tests you will find that this variable is not defined. Reason is that there is no such variable, it is special-cased in a number of places (and on Bram’s place I would make this a function which accepts a buffer number). I do not like the idea of special-casing it in yet another place, so proposed fix for those who need `b:changedtick` is (f69e7e51174b480b5c58779340d141486da1da0d)

``` Patch
commit f69e7e51174b480b5c58779340d141486da1da0d
Author: ZyX <kp-pav@yandex.ru>
Date:   Sat Jul 30 01:39:09 2016 +0300

    api/buffer: Add buffer_get_changedtick method

diff --git a/src/nvim/api/buffer.c b/src/nvim/api/buffer.c
index 55b535c..e8ea8e5 100644
--- a/src/nvim/api/buffer.c
+++ b/src/nvim/api/buffer.c
@@ -408,6 +408,9 @@ end:

 /// Gets a buffer-scoped (b:) variable.
 ///
+/// @note Use buffer_get_changedtick() for “variable” `b:changedtick`.
+///       This method only gets variables from b: dictionary.
+///
 /// @param buffer The buffer handle
 /// @param name The variable name
 /// @param[out] err Details of an error that may have occurred
@@ -423,6 +426,22 @@ Object buffer_get_var(Buffer buffer, String name, Error *err)
   return dict_get_value(buf->b_vars, name, err);
 }

+/// Gets a changed tick of a buffer
+///
+/// @param[in]  buffer  The buffer handle.
+///
+/// @return `b:changedtick` value.
+Integer buffer_get_changedtick(Buffer buffer, Error *err)
+{
+  const buf_T *const buf = find_buffer_by_handle(buffer, err);
+
+  if (!buf) {
+    return -1;
+  }
+
+  return buf->b_changedtick;
+}
+
 /// Sets a buffer-scoped (b:) variable
 ///
 /// @param buffer The buffer handle
diff --git a/test/functional/api/buffer_spec.lua b/test/functional/api/buffer_spec.lua
index cf8a83a..bb0418b 100644
--- a/test/functional/api/buffer_spec.lua
+++ b/test/functional/api/buffer_spec.lua
@@ -4,6 +4,7 @@ local clear, nvim, buffer = helpers.clear, helpers.nvim, helpers.buffer
 local curbuf, curwin, eq = helpers.curbuf, helpers.curwin, helpers.eq
 local curbufmeths, ok = helpers.curbufmeths, helpers.ok
 local funcs = helpers.funcs
+local neq = helpers.neq

 describe('buffer_* functions', function()
   before_each(clear)
@@ -243,6 +244,17 @@ describe('buffer_* functions', function()
       eq(1, funcs.exists('b:lua'))
       curbufmeths.del_var('lua')
       eq(0, funcs.exists('b:lua'))
+      curbuf('set_var', 'changedtick', true)
+      eq(true, curbuf('get_var', 'changedtick'))
+      neq(true, nvim('eval', 'b:changedtick'))
+    end)
+  end)
+
+  describe('get_changedtick', function()
+    it('works', function()
+      eq(2, curbufmeths.get_changedtick())
+      curbufmeths.set_lines(0, 1, false, {'abc\0', '\0def', 'ghi'})
+      eq(3, curbufmeths.get_changedtick())
     end)
   end)

```

Note: you _can_ have `b:` variable named `changedtick` using a number of methods: `setbufvar('%', 'changedtick', 1)`, `let b:['changedtick'] = 1`, etc. Including using `buffer_set_var` API method. It would not be available when evaluating `b:changedtick` or `getbufvar('%', 'changedtick')`, but other methods (`buffer_get_var` API call, `b:['changedtick']`, `b:.changedtick`) allow this.

**Comment by justinmk** [comment]
:+1:

**Comment by justinmk** [comment]
Closed by 0eab7ac4b98d9a29e9713251a0402b5fbc37355e
